gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+            display_fields=[f"info.{field}raw", f"info.{field}adj",],
+            verbose=verbose,
+        )
+
+    for subset in subsets:
+        for subfield in ["AC", "AN", "nhomalt"]:
+            # Check AC_raw >= AC adj
+            field = f"{subfield}{delimiter}{subset}{delimiter}"
+            generic_field_check(
+                t,
+                cond_expr=(t.info[f"{field}raw"] < t.info[f"{field}adj"]),
+                check_description=f"{field}raw >= {field}adj",
+                display_fields=[f"info.{field}raw", f"info.{field}adj",],
+                verbose=verbose,
+            )
+
+
+def frequency_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    subsets: List[str],
+    verbose: bool,
+    show_percentage_sites: bool = True,
+    delimiter: str = "-",
+) -> None:
+    """
+    Perform sanity checks on frequency data in input Table.
+
+    Check:
+        - Number of sites where gnomAD callset frequency is equal to a gnomAD subset frequency (both raw and adj)
+    
+    Also perform small spot checks:
+        - Counts total number of sites where the gnomAD allele count annotation is defined (both raw and adj)
+        
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] subsets: List of sample subsets.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param bool show_percentage_sites: If true, show the percentage of overall sites that fail; if False, show the number of sites that fail.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    for subset in subsets:
+        if subset != "":
+            subset += delimiter
+        else:
+            continue
+        for subfield in ["AC", "AN", "nhomalt"]:
+            logger.info("adj checks")
+            subfield_label = f"{subfield}{delimiter}adj"
+            subfield_subset_label = f"{subfield}{delimiter}{subset}adj"
+
+            generic_field_check(
+                t,
+                cond_expr=(t.info[subfield_label] == t.info[subfield_subset_label]),
+                check_description=f"{subfield_label} != {subfield_subset_label}",
+                display_fields=[
+                    f"info.{subfield_label}",
+                    f"info.{subfield_subset_label}",
+                ],
+                verbose=verbose,
+                show_percent_sites=show_percentage_sites,
+            )
+
+    freq_counts = t.aggregate(
+        hl.struct(
+            total_defined_AC=hl.agg.count_where(hl.is_defined(t.info["AC-adj"])),
+            total_defined_AC_raw=hl.agg.count_where(hl.is_defined(t.info["AC-raw"])),
+        )
+    )
+    logger.info(f"Frequency spot check counts: {freq_counts}")
+
+
+def sample_sum_check(
```

Comment:
There is already a `sample_sum_check` in `gnomad_methods` it looks like, can that function be generalized for our needs here?